### PR TITLE
Sync capabilities

### PR DESCRIPTION
**Pull request by marcserrat**

### CODE REVIEW

File: cnctcli/actions/products/configuration_values_sync.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            if scope_calc[2]:
+                payload['marketplace'] = {
+                    'id': scope_calc[2]
+                }
+
+            if data.action == 'update':
+                try:
+                    value = json.loads(data.value)
+                    payload['structured_value'] = value
+                except Exception:
```
Comment (ffaraoneim) [inline cnctcli/actions/products/configuration_values_sync.py:59]: Better to catch JSONDecodeError

File: cnctcli/actions/products/media_sync.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                continue
+            row_errors = self._validate_row(data)
+
+            if row_errors:
+                errors[row_idx] = row_errors
+                continue
+            if data.action == 'delete':
+                try:
+                    self._client.products[self._product_id].media[data.id].delete()
+                except ClientError:
```
Comment (ffaraoneim) [inline cnctcli/actions/products/media_sync.py:58]: Same here, If the ClientError is raised due to a timeout for example, the media has not been deleted but is counted as it was

File: cnctcli/actions/products/media_sync.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    self._update_sheet_row(ws, row_idx, media)
+                    updated_items.append(media)
+                else:
+                    media = self._client.products[self._product_id].media.create(
+                        data=payload,
+                        headers={'Content-Type': payload.content_type},
+                    )
+                    self._update_sheet_row(ws, row_idx, media)
+                    created_items.append(media)
+            except Exception as e:
```
Comment (ffaraoneim) [inline cnctcli/actions/products/media_sync.py:114]: Same here, better to catch ClientError

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: cnctcli/actions/products/actions_sync.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                    ].update(payload)
+                    updated_items.append(action)
+                except Exception as e:
+                    errors[row_idx] = [str(e)]
+
+            if data.action == 'create':
+                try:
+                    action = self._client.products[self._product_id].actions.create(payload)
+                    self._update_sheet_row(ws, row_idx, action)
+                    created_items.append(action)
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply cnctcli/actions/products/actions_sync.py:76]: not possible at this moment on time due backend issue

File: tests/conftest.py [hunk 5]
```diff
@@ -4,10 +4,13 @@
...
 import responses
 
+import os
+from shutil import copy2
+
 from tests.data import CONFIG_DATA
 from openpyxl import load_workbook
 
-@pytest.fixture()
+@pytest.fixture(scope='function')
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply tests/conftest.py:13]: For consistency

File: cnctcli/actions/products/actions_sync.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                "title": data.title,
+            }
+
+            if data.action == 'update':
+                try:
+                    action = self._client.products[self._product_id].actions[
+                        data.verbose_id
+                    ].update(payload)
+                    updated_items.append(action)
+                except Exception as e:
```
[the remark this answers is not in the document]
Comment (marcserrat) [reply cnctcli/actions/products/actions_sync.py:69]: i agree

File: cnctcli/actions/products/configuration_values_sync.py (hunk 1 above)
Comment (marcserrat) [reply cnctcli/actions/products/configuration_values_sync.py:59]: as discussed, due nature of excel that can handle cell as many types when is "general"

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**